src/services/Users.php

```diff
@@ -790,6 +791,20 @@ public function activateUser(User $user): bool
             return false;
         }
 
+        $user->active = true;
+        $user->pending = false;
+        $user->locked = false;
+        $user->suspended = false;
+        $user->verificationCode = null;
+        $user->verificationCodeIssuedDate = null;
+        $user->invalidLoginCount = null;
+        $user->lastInvalidLoginDate = null;
+        $user->lockoutDate = null;
+
+        if (!$user->validate()) {
+            return false;
+        }
+
```

Comment:
Validate up front, so we can bail if there is already an active user with this email/username.